File: 338.Counting_Bits/python/338.py
from typing import List

def countBits(self, n: int) -> List[int]:
    res = []
    count = int(0)

    while count < (n + 1):
        temp: int = count
        digit: int = 0

        while temp != 0:
            odd = temp % 2
            digit += odd
            temp = int(temp / 2)
        res.append(digit)
        count += 1

    
    return res

File: 338.Counting_Bits/C++/338.cpp
#include <vector>

using namespace std;

vector<int> countBits(int n) {
    int count = 0;
    vector<int> ans;
    while (count != (n + 1)) {
        int temp = count;
        int digit = 0;

        while (temp != 0) {
            int odd = temp % 2;
            digit += odd;
            temp = temp / 2;
        }
        ans.push_back(digit);
        count++;
    }

    return ans;
}